tests/pkasyncio_test.py

```
@@ -0,0 +1,118 @@
+    import asyncio
+
+    async def _all():
+        w = await _open()
+        pkdebug.pkdlog("open")
+        await _send(w)
+        pkdebug.pkdlog("send")
+        pkdebug.pkdlog("recv={}", await _recv(w))
+        return
+
+    async def _open():
+        from tornado import httpclient, websocket
+
+        return await websocket.websocket_connect(
+            httpclient.HTTPRequest(
+                url=f"ws://{_LOCALHOST}:{_PORT}{_URI}",
+            ),
+            ping_interval=100,
+            ping_timeout=1000,
+        )
+
+    async def _recv(ws):
+        return await ws.read_message()
+
+    async def _send(ws):
+        await ws.write_message("hello")
+
+    asyncio.run(_all())
+
+    # This does not work. The read_message hangs.
+    # try:
+    #     w = asyncio.run(_open())
+    #     pkdebug.pkdlog("open")
+    #     asyncio.run(_send(w))
+    #     pkdebug.pkdlog("send")
+    #     pkdebug.pkdlog("recv={}", asyncio.run(_recv(w)))
+    # except Exception as e:
+    #     pkdebug.pkdlog("exception={} stack={}", e, pkdebug.pkdexc())
+
+
+def _port():
+    import random
+
+    def _check_port(port):
+        import socket
+
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            s.bind((_LOCALHOST, int(port)))
+        return port
+
+    for p in random.sample(range(37000, 37100), 10):
+        try:
+            return _check_port(p)
+        except Exception:
+            pass
+    raise AssertionError(
+        f"ip={_LOCALHOST} unable to bind to port in range={37000-37100}"
+    )
+
+
+def _server():
+    from pykern import pkasyncio, pkdebug
+    from pykern.pkcollections import PKDict
+    from tornado import websocket
+
+    class _Echo(websocket.WebSocketHandler):
+        async def on_message(self, msg):
+            import asyncio
+
+            try:
+                pkdebug.pkdlog(msg)
+                self.write_message(msg)
+                pkdebug.pkdlog("send")
+            except Exception as e:
```

Why is this here? We swallow the error which looks odd to me.

---

Ah, thanks.